### DISCUSSION TITLE
Can't set trace params e.g session_id when using with llhama_index

### DISCUSSION THREAD

**Discussion by famoraes** [langfuse/langfuse#2527] [Support]
Hey there!

I've tried everything I found in docs/online regarding setting trace params to all traces in my application. Have found a few issues depending the way is setting. 

Currently my set up is using llhama_index, in my main fn I call:
```
llama_index.core.set_global_handler("langfuse")
langfuse_callback_handler: LlamaIndexCallbackHandler = cast(LlamaIndexCallbackHandler, llama_index.core.global_handler)
```
![Screenshot 2024-07-02 at 17 38 39](https://github.com/langfuse/langfuse/assets/4910050/0240a54b-7f59-4e7c-9219-dc10258f99f0)

This record traces correctly, setting name etc.

When I try to:
```langfuse_callback_handler.set_trace_params(session_id=my_id)```

This happens:
![Screenshot 2024-07-02 at 17 39 07](https://github.com/langfuse/langfuse/assets/4910050/920c4b09-034e-4b24-a47f-fdd812e86289)

I've tried set this way as well:
```llama_index.core.set_global_handler("langfuse", session_id=my_id)```

I tried to set a root trace, set the callback handler using llhama_index `Settings` etc. None worked.
Thank you!

**Comment by hassiebp** [comment]
Thanks for your report, @famoraes! Could you please share a minimum repro example for the second case where setting params fails? That will help to pinpoint the issue.

**Comment by famoraes** [comment]
```
import llama_index.core
from langfuse.llama_index import LlamaIndexCallbackHandler


def test_file_summary(session_id):
    llama_index.core.set_global_handler("langfuse")
    langfuse_callback_handler: LlamaIndexCallbackHandler = cast(LlamaIndexCallbackHandler, llama_index.core.global_handler)
    langfuse_callback_handler.set_trace_params(session_id=session_id)

    # my llm code, I basically create a KnowledgeGraphIndex based on some files, and I'm using langchain StructuredOutputParser and 
    # ResponseSchema to parse my responses.  Nothing especial there.
    # then I query against it e.g document_query.query(...)
   
    langfuse_callback_handler.flush()
```

I can share the inner code if it would be helpful, it's around 200 lines.
Thank you!

**Reply by hassiebp** [reply]
Hi @famoraes - I suspect something is going wrong in your implementation by trying to access to LlamaIndex global handler. Could you try using the alternative approach by registering the callback handler in Settings.callback_manager as described here? https://langfuse.com/docs/integrations/llama-index/get-started#set-trace-params

``` python
from llama_index.core import Settings
from llama_index.core.callbacks import CallbackManager
from langfuse import langfuse
 
# Instantiate a new LlamaIndexCallbackHandler and register it in the LlamaIndex Settings
langfuse_handler = LlamaIndexCallbackHandler()
Settings.callback_manager = CallbackManager([langfuse_handler])
 
def my_func():
  # Set trace parameters before executing your LlamaIndex code
  langfuse_callback_handler.set_trace_params(
    user_id="user-123",
    session_id="session-abc",
    tags=["production"]
  )
 
  # Your LlamaIndex code, trace will include the set parameters
```

**Reply by famoraes** [reply]
Hey @hassiebp, thank you for your reply!
I just tried that way, didn't work either. Although I'm not quite sure where `langfuse_callback_handler` is set. I suspected it was a typo and the right variable is the `langfuse_handler`. So in my core I called:
```python
langfuse_handler = LlamaIndexCallbackHandler()
Settings.callback_manager = CallbackManager([langfuse_handler])
 
def my_func():
  # Set trace parameters before executing your LlamaIndex code
  langfuse_handler.set_trace_params(
    user_id="user-123",
    session_id="session-abc",
    tags=["production"]
  )
```
